### DISCUSSION TITLE
Nuxt Directus SDK plugin

### DISCUSSION THREAD

**Discussion by matt-rolley** [directus-labs/agency-os#33] [General]
Hey guys,

Noticed you'd wrote a custom plugin for the custom auth/setting up Directus within Nuxt and I've started a similar project (https://github.com/rolleyio/nuxt-directus-sdk and nuxt-directus-sdk on npm) which does a lot of the same features, but also automatically generates your Directus collections types for use inside Nuxt, adds the admin panel to the Nuxt devtools and sets up Directus for use in server routes too.

I'm hoping to add quite a few more features in the near future, but wondered if it's worth combining efforts? The docs will be following shortly.

No worries if not, just a thought!

**Comment by bryantgillespie** [comment]
@matt-rolley Let's chat about this for sure.

**Comment by kheiner** [comment]
@matt-rolley This project looks very impressive so far - I was mid-process of trying to figure out how to improve my workflow with the directus SDK and stumbled here.  I really like the simplicity of the dev proxy on top of all of the other features.

@bryantgillespie Matt's project looks polished and may be worth working internally with the Directus folks to sanction/audit the module at this point.  I was watching a stream by Daniel Roe [where he attempted to use the existing module posted on nuxt and had to back-track because it didn't meet his needs.](https://www.youtube.com/live/zlLJVydoP4o?si=lf7lUcskn5o__Gr0&t=1161) Totally recognize that the existing module is also user created and not to diminish the work there, but it does now appear to be an abandoned project.

**Comment by matt-rolley** [comment]
@kheiner Thanks for the kind works! There are a still few rough edges, but it covers our needs 95% of the time and we try to actively keep on top of new Directus features/releases, like the Visual Editor integration etc.

I'm always up for chatting through the plugin for future ideas or just general feedback. If there is anything at all, please do let me know (or feel free to leave issues/feedback in the repo) - I'm always happy to try to help where I can!

Regarding Directus sanctioning/auditing the plugin, I'd be more than happy with that too! Happy to jump on a call anytime.
